src/Orchard.Web/Modules/Orchard.Widgets/Services/DefaultLayerEvaluationService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using Orchard.Localization;
+using Orchard.Logging;
+using Orchard.Widgets.Models;
+using Orchard.ContentManagement;
+
+namespace Orchard.Widgets.Services{
+    public class DefaultLayerEvaluationService : ILayerEvaluationService {
+        private readonly IRuleManager _ruleManager;
+        private readonly IOrchardServices _orchardServices;
+
+        public DefaultLayerEvaluationService(IRuleManager ruleManager, IOrchardServices orchardServices) {
+            _ruleManager = ruleManager;
+            _orchardServices = orchardServices;
+
+            Logger = NullLogger.Instance;
+            T = NullLocalizer.Instance;
+        }
+
+        public ILogger Logger { get; set; }
+        public Localizer T { get; private set; }
+
+        /// <summary>
+        /// Retrieves every Layer from the Content Manager and evaluates each one.
+        /// </summary>
+        /// <returns>
+        /// A collection of integers that represents the Ids of each active Layer
+        /// </returns>
+        public IEnumerable<int> GetActiveLayerIds(){
+            // Once the Rule Engine is done:
+            // Get Layers and filter by zone and rule
+            // NOTE: .ForType("Layer") is faster than .Query<LayerPart, LayerPartRecord>()
+            var activeLayers = _orchardServices.ContentManager.Query<LayerPart>().ForType("Layer").List();
```

Comment:
I purposely have not placed the layers in `ICacheManager` here because deployments into multi node instances will require a distributed cache. If I used a distributed cache (`ICacheService`), then that would make `Orchard.Widgets` become dependant on `Orchard.Caching`, which I think we should avoid.

That said, I have created an implementation of this service in our internal repo that takes the layers from `ICacheService`, and we are seeing relatively large performance gains (especially on sites with many layers). I'd happily contribute this back, but I think it'd be best to create another PR once this one has been merged.